File: php-laravel/lib/app/Models/RequestTimeFilterPostcodeDistrictsDepartureSearch.php
<?php
/**
 * RequestTimeFilterPostcodeDistrictsDepartureSearch
 */
namespace app\Models;

/**
 * RequestTimeFilterPostcodeDistrictsDepartureSearch
 */
class RequestTimeFilterPostcodeDistrictsDepartureSearch {

    /** @var string $id */
    private $id;

    /** @var \app\Models\RequestTransportation $transportation */
    private $transportation;

    /** @var int $travelTime */
    private $travelTime;

    /** @var \DateTime $departureTime */
    private $departureTime;

    /** @var double $reachablePostcodesThreshold */
    private $reachablePostcodesThreshold;

    /** @var \app\Models\RequestTimeFilterPostcodeDistrictsProperty[] $properties */
    private $properties;

    /** @var \app\Models\RequestRangeFull $range */
    private $range;

}

File: scalaz/src/main/scala/org/openapitools/client/api/RequestTimeFilterPostcodeDistrictsDepartureSearch.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import org.joda.time.DateTime

import RequestTimeFilterPostcodeDistrictsDepartureSearch._

case class RequestTimeFilterPostcodeDistrictsDepartureSearch (
  id: String,
transportation: RequestTransportation,
travelTime: Integer,
departureTime: DateTime,
reachablePostcodesThreshold: Double,
properties: List[RequestTimeFilterPostcodeDistrictsProperty],
range: Option[RequestRangeFull])

object RequestTimeFilterPostcodeDistrictsDepartureSearch {
  import DateTimeCodecs._

  implicit val RequestTimeFilterPostcodeDistrictsDepartureSearchCodecJson: CodecJson[RequestTimeFilterPostcodeDistrictsDepartureSearch] = CodecJson.derive[RequestTimeFilterPostcodeDistrictsDepartureSearch]
  implicit val RequestTimeFilterPostcodeDistrictsDepartureSearchDecoder: EntityDecoder[RequestTimeFilterPostcodeDistrictsDepartureSearch] = jsonOf[RequestTimeFilterPostcodeDistrictsDepartureSearch]
  implicit val RequestTimeFilterPostcodeDistrictsDepartureSearchEncoder: EntityEncoder[RequestTimeFilterPostcodeDistrictsDepartureSearch] = jsonEncoderOf[RequestTimeFilterPostcodeDistrictsDepartureSearch]
}
